File: Fruit.rb
class Fruit
	@@longestName = ""
	def self.getLongestName
		@@longestName
	end
	def initialize(name, colour)
		@name = name
		if name.length() > @@longestName.length()
			@@longestName = name
		end
		@colour = colour
	end
	def name
		@name
	end
	def colour=(colour)
		@colour = colour
	end
	def describe()
		"I am a " + @colour + " fruit called " + @name + "."
	end
end
print "Count down from what number? "
start = gets
for i in start.to_i.downto(1)
	print i.to_s + ", "
end
puts "lift off!"
fruits = [ Fruit.new("Apple", "Green"),
			Fruit.new("Strawberry", "Red"),
			Fruit.new("Orange", "Orange"),
			Fruit.new("Banana", "Yellow"),
			Fruit.new("Grape", "Green")]
fruitCount = fruits.length
puts "There are " + fruitCount.to_s + " fruits..."
for fruit in fruits
	if fruit.name == "Grape"
		fruit.colour = "Purple"
	end
	puts fruit.describe
end
puts "The longest fruit name is " + Fruit.getLongestName() + "!"

File: Fruit.fs
open System
type Fruit(name: string, colour) =
    static let mutable longestName = ""
    do longestName <- if name.Length > longestName.Length then name else longestName
    static member getLongestName() = longestName
    member this.name = name
    member val colour = colour with get, set 
    member this.describe() = sprintf "I am a %s fruit called %s" this.colour this.name
let main() = 
    printf "Count down from what number? "
    let start = Console.ReadLine() |> int
    for i = start downto 1 do
        printf "%d, " i
    printfn "lift off!"
    let fruits = [ Fruit("Apple", "Green");
                   Fruit("Strawberry", "Red");
                   Fruit("Orange", "Orange");
                   Fruit("Banana", "Yellow");
                   Fruit("Grape", "Green") ]
    let fruitCount = fruits.Length
    printfn "There are %d fruits..." fruitCount
    fruits |> List.iter (fun fruit -> 
        match fruit.name with
        | "Grape" -> fruit.colour <- "Purple"
        | _ -> ()
        printfn "%s" (fruit.describe())
    )
    printfn "The longest fruit name is %s!" (Fruit.getLongestName())
main()
